code/datums/status_effects/xeno_buffs.dm

```diff
@@ -24,6 +27,315 @@
+	buff_owner.adjust_sunder(-1)
+	return ..()
+
+// ***************************************
+// *********** Enhancement
+// ***************************************
+/particles/drone_enhancement
+	icon = 'icons/effects/particles/generic_particles.dmi'
+	icon_state = "fire"
+	color = "#9828CC"
+	width = 100
+	height = 100
+	count = 1000
+	spawning = 3
+	lifespan = 9
+	fade = 12
+	grow = 0.04
+	velocity = list(0, 0)
+	position = generator("circle", 16, 16, NORMAL_RAND)
+	drift = generator("vector", list(0, -0.15), list(0, 0.15))
+	gravity = list(0, 0.8)
+	scale = generator("vector", list(0.1, 0.1), list(0.6,0.6), NORMAL_RAND)
+	rotation = 0
+	spin = generator("num", 10, 20)
+
+/datum/status_effect/drone_enhancement
+	id = "drone enhancement"
+	duration = -1
+	tick_interval = 2 SECONDS
+	alert_type = null
+	/// Used to track the giver of this buff.
+	var/mob/living/carbon/xenomorph/buffing_target
+	/// Used to track the owner of this buff.
+	var/mob/living/carbon/xenomorph/buff_owner
+	/// Used for particles. Holds the particles instead of the mob. See particle_holder for documentation.
+	var/obj/effect/abstract/particle_holder/particle_holder
+	/// References the Essence Link action and its vars.
+	var/datum/action/xeno_action/activable/essence_link/essence_link_action
+	/// References the Enhancement action and its vars.
+	var/datum/action/xeno_action/enhancement/enhancement_action
+	/// If the target xeno was within range.
+	var/was_within_range = TRUE
+	/// The plasma cost per tick of this ability.
+	var/plasma_cost
+
+/datum/status_effect/drone_enhancement/on_creation(mob/living/new_owner, mob/living/carbon/new_target)
+	buff_owner = new_owner
+	buffing_target = new_target
+	essence_link_action = buffing_target.actions_by_path[/datum/action/xeno_action/activable/essence_link]
+	enhancement_action = buffing_target.actions_by_path[/datum/action/xeno_action/enhancement]
+	plasma_cost = round(buffing_target.xeno_caste.plasma_max * 0.1)
+	RegisterSignal(buff_owner, COMSIG_MOVABLE_MOVED, .proc/handle_dist)
+	RegisterSignal(buffing_target, COMSIG_MOVABLE_MOVED, .proc/handle_dist)
+	RegisterSignal(buff_owner, COMSIG_MOB_DEATH, .proc/handle_death)
+	RegisterSignal(buffing_target, COMSIG_MOB_DEATH, .proc/handle_death)
+	INVOKE_ASYNC(buff_owner, /mob/living/carbon/xenomorph.proc/emote, "roar2")
+	toggle_buff(TRUE)
+	return ..()
+
+/datum/status_effect/drone_enhancement/on_remove()
+	buff_owner.balloon_alert(buff_owner, "Enhancement inactive")
+	buffing_target.balloon_alert(buffing_target, "Enhancement inactive")
+	UnregisterSignal(buff_owner, list(COMSIG_MOVABLE_MOVED, COMSIG_MOB_DEATH))
+	UnregisterSignal(buffing_target, list(COMSIG_MOVABLE_MOVED, COMSIG_MOB_DEATH))
+	toggle_buff(FALSE)
+	return ..()
+
+/datum/status_effect/drone_enhancement/tick()
+	if(buffing_target.plasma_stored < enhancement_action.plasma_cost)
+		enhancement_action.end_ability()
+		return
+	buffing_target.use_plasma(plasma_cost)
```

Comment:
Why are you checking threshold on enhancement_action's plasma cost but using the status effect's set amount?